### PR TITLE
Modern Ruby + ActiveRecord, Update Travis Config

### PR DESCRIPTION
**Pull request by joshbranham**
In an effort to make forward progress, and keep Travis credits from getting burned up too quickly, this modifies the configuration to:

- Stop testing EOL rubies
- Stop testing EOL ActiveRecord, except 5.1 and 5.2
- Remove checks for older removed ActiveRecord versions
- Include a fix for ActiveRecord 5.2, from fork source https://github.com/priyankatapar/attr_encrypted/commit/7e8702bd5418c927a39d8dd72c0adbea522d5663

With the above tests are now passing and this is ready for review so follow on work like adding Rails 7 support can continue.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,59 +1,17 @@
...
-  - ACTIVERECORD=3.1.0
-  - ACTIVERECORD=3.2.0
-  - ACTIVERECORD=4.0.0
-  - ACTIVERECORD=4.1.0
-  - ACTIVERECORD=4.2.0
-  - ACTIVERECORD=5.0.0
-  - ACTIVERECORD=5.1.1
-matrix:
-  exclude:
-    - rvm: 2.0
```
Comment (mvastola) [inline .travis.yml:23]: Were we seriously testing different versions of RVM as part of the build matrix? :laughing: 

Good catch

**Review by mvastola — CHANGES_REQUESTED** [review]
Just a few questions to make sure we get this right..

Also, any reason why the build is currently failing? Hadn't the most recent commit been passing?

**Comment by mvastola** [comment]
> Also, any reason why the build is currently failing? Hadn't the most recent commit been passing?

Oh, is it just because you added Rails 7?

File: .travis.yml [hunk 2]
```diff
@@ -1,59 +1,17 @@
-sudo: false
```
[the remark this answers is not in the document]
Comment (joshbranham) [reply .travis.yml:1]: We don't need it, we were using the default value. The Travis UI had a list of config options to change/remove and this was one of them.

File: .travis.yml [hunk 3]
```diff
@@ -1,59 +1,17 @@
...
 language: ruby
+dist: bionic
```
[the remark this answers is not in the document]
Comment (joshbranham) [reply .travis.yml:2]: It should probably be `focal`, the list is here. https://docs.travis-ci.com/user/reference/overview/#the-amd64-builds

File: .travis.yml (hunk 1 above)
Comment (joshbranham) [reply .travis.yml:23]: I think that was just telling Travis to not build that ruby version + activerecord version. In this case they are all super old and we are no longer testing them as suggested in this branch, so I removed them.

File: .travis.yml [hunk 4]
```diff
@@ -1,59 +1,17 @@
...
-  - 2.0
-  - 2.1
-  - 2.2.2
-  - 2.3.0
-  - 2.4.0
-  - 2.5.0
-  - rbx
+  - 2.5.9
+  - 2.6.10
+  - 2.7
```
Comment (joshbranham) [inline .travis.yml:7]: I changed these as requested and rvm is _not_ installing the latest. For that reason I suggest just putting the latest patch back in the file for now.

https://app.travis-ci.com/github/attr-encrypted/attr_encrypted/jobs/585022565

```
Required ruby-2.7.1 is not installed.
Creating alias 2.7 for ruby-2.7.1....
```

**Comment by joshbranham** [comment]
> > Also, any reason why the build is currently failing? Hadn't the most recent commit been passing?
> 
> Oh, is it just because you added Rails 7?

I haven't added rails 7 to this yet, as it will fail too, but I can if you think. My thought was to add it in a separate PR where I can also try and get some of the existing tests to pass.

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mvastola) [reply .travis.yml:1]: Got it. Just wanted to make sure either it was the default or there was a good reason. :+1:

File: .travis.yml (hunk 1 above)
Comment (mvastola) [reply .travis.yml:23]: Oh, so `rvm` here means `ruby`?

Comment (joshbranham) [reply .travis.yml:23]: Yeah per https://docs.travis-ci.com/user/languages/ruby/#specifying-ruby-versions-and-implementations

**Comment by mvastola** [comment]
> As far as I can tell the build has been failing for at least 4 years now, as shown here https://app.travis-ci.com/github/attr-encrypted/attr_encrypted/builds
> 
> The last successful, per Travis, was this commit [11df93a](https://github.com/attr-encrypted/attr_encrypted/commit/11df93aef14c661dd0c03169d382a0412f93124e).

@joshbranham, yes, but the sole commit to `master` since then (AFAIK) was updating `README.md` to announce the need for maintainers.. :slightly_frowning_face: 

> I am happy to start fixing things to get some of the builds passing, but my hunch says small changes here for historical context is better.

If we can retrace our steps easily (as you point out, this isn't a big commit) re: the changes and thereby keep things passing, I think that's preferable. That said, I'm not opposed to merging if that's not possible (or if you think doing so would improve our work(flow) in some way).

Lmk what you think.

File: .travis.yml (hunk 4 above)
Comment (mvastola) [reply .travis.yml]: Hrm. Then yeah, we'll have to fully specify them unfortunate;y. (The goal was to make sure we used the latest versions, and clearly this suggestion does the opposite.) That's annoying.

**Comment by joshbranham** [comment]
@mvastola this has been updated as requested, we are now testing modern Ruby and ActiveRecord and passing with existing functionality. The only things I removed were the old versions, as discussed, and conditionals in the code that were checking old versions.

Additionally, ActiveRecord 5.2 support was broken. A contributor added support in their fork to resolve this https://github.com/priyankatapar/attr_encrypted/commit/7e8702bd5418c927a39d8dd72c0adbea522d5663

I think it might be worth reaching out to them to see if they want to open a pull adding that, either to master or my branch. What are your thoughts? It has been 3 years however.

**Review by mvastola — APPROVED** [review]
This is PERFECT! Thanks!

**Pull request merged**